packages/mui-material/src/useAutocomplete/useAutocomplete.js

```diff
@@ -137,14 +144,21 @@ function useAutocomplete(props) {
   const defaultHighlighted = autoHighlight ? 0 : -1;
   const highlightedIndexRef = React.useRef(defaultHighlighted);
 
+  // Caculate the initial inputValue only on first render (mount)
+  // Ensures that the initialInputValue remains constant during the component's lifetime
+  // defaultValue does not need to dynamically affect the inputValue after initialization
+  const [initialInputValue] = React.useState(() =>
+    getInitialInputValue(defaultValue, multiple, getOptionLabel),
+  );
```

Comment:
Would a `useRef` be more fitting here?

Reply:
I considered that, but now when I think more the `useState` initialization function is also not needed. Since the `defaultValue` for the Autocomplete won't change dynamically (either externally by developers or within the component lifecycle), we can directly calculate and pass it as the default input value: https://github.com/mui/material-ui/pull/44873/commits/d45ad3b2a561da692c1fbe679d8b6510205092e8

Reply:
I think this might lead to confusing errors though:

- If a user updates `getOptionLabel`, for example, the `default` provided to `useControlled` would change
- The user would get to this warning
  https://github.com/mui/material-ui/blob/1d38c20a1dcd4090df9d85cfcb4425bfcec093cb/packages/mui-utils/src/useControlled/useControlled.js#L38
- They will think the issue is with `inputValue` when really the issue is about changing `getOptionLabel`

Reply:
You're right. I've also switched it back to using `ref` instead of `useState`. `useRef` feels more appropriate here since the value needs to persist across re-renders without causing updates and is only used on mount. Although previously with `useState` I wasn't using the `setState` method which causes re-rendering, this approach feels cleaner and more intentional. What do you think?